frontend/catalyst/api_extensions/function_maps.py

```diff
@@ -30,132 +33,79 @@
 from catalyst.tracing.contexts import EvaluationContext
 
 
-## API ##
-def vmap(
-    fn: Callable,
-    in_axes: Union[int, Sequence[Any]] = 0,
-    out_axes: Union[int, Sequence[Any]] = 0,
-    axis_size: Optional[int] = None,
-) -> Callable:
-    """A :func:`~.qjit` compatible vectorizing map.
-    Creates a function which maps an input function over argument axes.
+@dataclass
+class VmapParams:
+    """Common vmap parameters. The parameters are expected to be checked before the creation of
+    this structure"""
 
-    Args:
-        f (Callable): A Python function containing PennyLane quantum operations.
-        in_axes (Union[int, Sequence[Any]]): Specifies the value(s) over which input
-            array axes to map.
-        out_axes (Union[int, Sequence[Any]]): Specifies where the mapped axis should appear
-            in the output.
-        axis_size (int): An integer can be optionally provided to indicate the size of the
-            axis to be mapped. If omitted, the size of the mapped axis will be inferred from
-            the provided arguments.
+    in_axes: Union[int, Sequence[Any]]
+    out_axes: Union[int, Sequence[Any]]
+    axis_size: Optional[int]
```

Comment:
Just curious about the choice to introduce a dataclass. There is nothing really wrong with it, although it does introduce an extra level of indirection for only three parameters.

Reply:
I've just tried to follow that pattern as well, in case we need to add more parameters in the future. But I agree, we can simplify it by now. Done.